psetv2/updater.go

```diff
@@ -249,52 +249,50 @@ func (arg AddInIssuanceArgs) validate() error {
 			return fmt.Errorf("invalid issuance token address: %s", err)
 		}
 	}
-	if !arg.matchAddressTypes() {
-		return ErrInIssuanceAddressesMismatch
-	}
 
 	return nil
 }
 
-func (arg AddInIssuanceArgs) matchAddressTypes() bool {
-	if len(arg.TokenAddress) <= 0 {
-		return true
-	}
-
-	a, _ := address.IsConfidential(arg.AssetAddress)
-	b, _ := address.IsConfidential(arg.TokenAddress)
-	return a == b
-}
-
 func (arg AddInIssuanceArgs) tokenFlag() uint {
 	if arg.BlindedIssuance {
 		return uint(ConfidentialReissuanceTokenFlag)
 	}
 	return uint(NonConfidentialReissuanceTokenFlag)
 }
 
-func (arg AddInIssuanceArgs) parseAssetAddress() ([]byte, []byte) {
-	script, _ := address.ToOutputScript(arg.AssetAddress)
-	isConf, _ := address.IsConfidential(arg.AssetAddress)
+func parseAddress(addr string) ([]byte, []byte) {
+	script, _ := address.ToOutputScript(addr)
+	isConf, _ := address.IsConfidential(addr)
 	if !isConf {
 		return script, nil
 	}
-	info, _ := address.FromConfidential(arg.AssetAddress)
+	info, _ := address.FromConfidential(addr)
 	return info.Script, info.BlindingKey
 }
 
+func (arg AddInIssuanceArgs) parseAssetAddress() ([]byte, []byte) {
+	return parseAddress(arg.AssetAddress)
+}
+
 func (arg AddInIssuanceArgs) parseTokenAddress() ([]byte, []byte) {
-	script, _ := address.ToOutputScript(arg.TokenAddress)
-	isConf, _ := address.IsConfidential(arg.TokenAddress)
-	if !isConf {
-		return script, nil
+	return parseAddress(arg.TokenAddress)
+}
```

Comment:
either we remove these or add also for reissuance args, but let's fix this code inconsistency.